### PR TITLE
workspace creation endpoint

### LINKED ISSUE #31: Create endpoint for workspace: /api/v1/create-new-workspace
Get customer id from the request context

### PR DESCRIPTION
**Pull request by Nemadede**

### CODE REVIEW

File: app/controller/workspace/workspace.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+from fastapi import APIRouter, Depends, Body
+
+from app.config.authorization import Authorization
+from app.config.success_response import CustomException, SuccessResponse
+from app.dto.model.workspace import CreateWorkspaceDTO
+
+router = APIRouter(dependencies=[Depends(Authorization.authenticate)])
```
Comment (Real-Music) [inline app/controller/workspace/workspace.py]: Create a tag and call it workspace

File: app/controller/workspace/workspace.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    from app.service.model.workspace import WorkspaceLib
+    from app.service.model.language import LanguageLib
+    from app.dto.model.workspace import WorkspaceDTO
+    from fastapi.encoders import jsonable_encoder
+    from fastapi import status
+
+    # check default language if exist
+    language = LanguageLib.find_by(where={"name": data.default_language})
+    if not language:
+        return SuccessResponse(data={}).set_message("language not available for now").\
```
Comment (Real-Music) [inline app/controller/workspace/workspace.py]: data = None

File: app/controller/workspace/workspace.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    record["customer_id"] = customer.id
+    record["customer_count"] = 1
+
+    # create workspace
+    workspace: WorkspaceDTO = WorkspaceLib.create(data=record)
+
+    # create workspace detail
+    workspace_detail = WorkspaceLib.create_workspace_detail(data={"workspace_id": workspace.id,
+                                                                  "customer_id": customer.id,
+                                                                  "workspace_role_id": 1}
```
Comment (Real-Music) [inline app/controller/workspace/workspace.py]: what's this here

File: app/dto/model/workspace.py [hunk 4]
```diff
@@ -22,3 +24,11 @@ class WorkspaceDTOs(BaseModel):
 
     class Config:
         orm_mode = True
+
```
Comment (Real-Music) [inline app/dto/model/workspace.py:26]: This should be remove from here

File: app/models/workspace.py [hunk 5]
```diff
@@ -12,6 +12,7 @@ class Workspace(Base_Model):
     name = Column(String(255), nullable=False)
     customer_id = Column(Integer, ForeignKey("customer.id", ondelete="RESTRICT"), nullable=True)
     customer_count = Column(Integer, default=0)
+    default_language = Column(Integer, ForeignKey("language.id", ondelete="NO ACTION"), nullable=True)
```
Comment (Real-Music) [inline app/models/workspace.py]: nullable here should be false

File: app/models/workspace_detail.py [hunk 6]
```diff
@@ -10,7 +10,7 @@ class WorkspaceDetail(Base_Model):
     id = Column(Integer, primary_key=True, autoincrement=True)
 
     workspace_id = Column(Integer, ForeignKey("workspace.id", ondelete="CASCADE"), nullable=True)
-    customer_id = Column(Integer, ForeignKey("customer.id", ondelete="RESTRICT"), nullable=True)
+    customer_id = Column(Integer, ForeignKey("customer.id", ondelete="CASCADE"), nullable=True)
```
Comment (Real-Music) [inline app/models/workspace_detail.py]: Pay attention to details

File: app/service/model/workspace.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        workspace = Workspace(**data)
+
+        db.add(workspace)
+        db.flush()
+
+        return WorkspaceDTO.from_orm(workspace)
+
+    @staticmethod
+    @session_hook
+    def create_workspace_detail(db: Session, data: dict) -> WorkspaceDetailDTO:
```
Comment (Real-Music) [inline app/service/model/workspace.py]: This should be removed from here

**Review by Real-Music — CHANGES_REQUESTED** [review]

File: app/controller/workspace/workspace.py (hunk 1 above)
Comment (Nemadede) [reply app/controller/workspace/workspace.py]: what's a tag?

**Review by Real-Music — APPROVED** [review]

**Pull request merged**